core/heap.c

```diff
@@ -1901,8 +1901,13 @@ vmm_heap_init()
     if (DYNAMO_OPTION(vm_reserve)) {
         vmm_heap_unit_init(&heapmgt->vmcode, DYNAMO_OPTION(vm_size), true, "vmcode");
         if (!REACHABLE_HEAP()) {
-            vmm_heap_unit_init(&heapmgt->vmheap, DYNAMO_OPTION(vmheap_size), false,
-                               "vmheap");
+            if (DYNAMO_OPTION(inject_x64)){
```

Comment:
This should also test for this being a WOW64 process.  This option would be set in 64-bit parents too so by itself it does not tell you it's WOW64.

Reply:
No idea about how to do that without messing with function's arguments, so I left a message here for now.

Reply:
Use `is_wow64_process(NT_CURRENT_PROCESS)`

Reply:
Fixed